server/controllers/auth.js

```diff
@@ -1,43 +1,91 @@
 
 module.exports = {
-    postLogin: (req, res, next) => {
-        const validationErrors = [];
-        if (!validator.isEmail(req.body.email))
-          validationErrors.push({ msg: 'Please enter a valid email address.' });
-        if (validator.isEmpty(req.body.password))
-          validationErrors.push({ msg: 'Password cannot be blank.' });
-        if (validationErrors.length) {
-          req.flash('errors', validationErrors);
-          return res.redirect('/login');
+  postLogin: (req, res, next) => {
+    const validationErrors = [];
+    if (!validator.isEmail(req.body.email))
+      validationErrors.push({ msg: 'Please enter a valid email address.' });
+    if (validator.isEmpty(req.body.password))
+      validationErrors.push({ msg: 'Password cannot be blank.' });
+    if (validationErrors.length) {
+      req.flash('errors', validationErrors);
+      return res.redirect('/login');
+    }
+    req.body.email = validator.normalizeEmail(req.body.email, {
+      gmail_remove_dots: true,
+    });
+
+    passport.authenticate('local', (err, user, info) => {
+      if (err) {
+        return next(err);
+      }
+      if (!user) {
+        req.flash('errors', info);
+        return res.redirect('/login');
+      }
+      req.logIn(user, (err) => {
+        if (err) {
+          return next(err);
         }
-        req.body.email = validator.normalizeEmail(req.body.email, {
-          gmail_remove_dots: false,
-        });
-
-        passport.authenticate('local', (err, user, info) => {
-          if (err) {
-            return next(err);
-          }
-          if (!user) {
-            req.flash('errors', info);
-            return res.redirect('/login');
-          }
-          req.logIn(user, (err) => {
-            if (err) {
-              return next(err);
-            }
-            req.flash('success', { msg: 'Success! You are logged in.' });
-            // res.redirect(req.session.returnTo || '/');
-            res.json({login: 'successful'});
-          });
-        })(req, res, next);
-    },
-
-    postSignUp: (res, req, next) => {
-      console.log(res, req, next());
+        req.flash('success', { msg: 'Success! You are logged in.' });
+        // res.redirect(req.session.returnTo || '/');
+        res.json({login: 'successful'});
+      });
+    })(req, res, next);
+  },
+
+  postSignUp: async (req, res, next) => {
+    const validationErrors = [];
+    const {username, password, confirmPassword} = req.body;
+    let {email} = req.body;
+
+    if (!validator.isEmail(email)){
+      validationErrors.push({ msg: 'Please enter a valid email address.'});
+    }
+    if (!validator.isStrongPassword(password)){
+      validationErrors.push({ msg: 'Password must be atleast 8 characters long, Contain atleast 1 lowercase letter, 1 uppercase letter, 1 number and 1 symbol.'});
+    }
+    if (password !== confirmPassword){
+      validationErrors.push({ msg: 'Passwords do not match.'});
+    }
```

Comment:
This logic should be implemented client side, not server side. The frontend can use JavaScript to ensure form fields match before allowing the user to submit. API clients can just send a password directly.